packages/content/src/portal.ts

```diff
@@ -16,10 +16,61 @@ import {
   getItemDataUrl,
   getItemThumbnailUrl,
   cloneObject,
-  includes
+  includes,
+  failSafe
 } from "@esri/hub-common";
 import { getContentMetadata } from "./metadata";
 
+function getGroupIds(
+  itemId: string,
+  requestOptions: IHubRequestOptions
+): Promise<string[]> {
+  return getItemGroups(itemId, requestOptions).then(response => {
+    const { admin, member, other } = response;
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+}
+
+// simultaneously execute multiple async requests to populate content properties
+// returns a hash of all the resolved property values that also has
+// an array of property names for which the request failed
+// NOTE: assumes each request takes only the item id and request options
+function fetchProperties(
```

Comment:
this function makes it easy to add more fetches for properties like `server` that we'll need to populate for datasets.

Reply:
Nice API design.